rasa/nlu/training_data/training_data.py

```diff
@@ -75,12 +75,16 @@ def merge(self, *others: "TrainingData") -> "TrainingData":
             nlg_stories,
         )
 
-    def filter_by_intent(self, intent: Text):
```

Comment:
Since this is not a private method, should we deprecate it with a warning and call the new method with an appropriate lambda function?
Since we still support previous minor versions, someone's custom code may break because of this.

Reply:
👍